user-operator/src/main/java/io/strimzi/operator/user/model/acl/SimpleAclRule.java

```diff
@@ -133,13 +136,22 @@ public static SimpleAclRule fromAclBinding(AclBinding aclBinding) {
     }
 
     /**
-     * Creates SimpleAclRule object based on AclRule object which is received as part ofthe KafkaUser CRD.
+     * Creates SimpleAclRule object based on AclRule object which is received as part of the KafkaUser CRD.
      *
-     * @param rule  AclRule object from KafkaUser CR
+     * @param rule AclRule object from KafkaUser CR
      * @return The SimpleAclRule.
      */
-    public static SimpleAclRule fromCrd(AclRule rule)   {
-        return new SimpleAclRule(rule.getType(), SimpleAclRuleResource.fromCrd(rule.getResource()), rule.getHost(), rule.getOperation());
+    @SuppressWarnings("deprecation")
+    public static List<SimpleAclRule> fromCrd(AclRule rule) {
+        if (rule.getOperations() != null) {
+            List<SimpleAclRule> simpleAclRules = new ArrayList<>();
+            for (AclOperation operation : rule.getOperations()) {
+                simpleAclRules.add(new SimpleAclRule(rule.getType(), SimpleAclRuleResource.fromCrd(rule.getResource()), rule.getHost(), operation));
+            }
+            return simpleAclRules;
+        } else {
+            return Collections.singletonList(new SimpleAclRule(rule.getType(), SimpleAclRuleResource.fromCrd(rule.getResource()), rule.getHost(), rule.getOperation()));
+        }
```

Comment:
You should probably throw some error (`InvalidResourceException`) in case both are set.